tests/30rooms/13anonymousaccess.pl

```diff
@@ -234,6 +236,88 @@
       })
    };
 
+test "Anonymous users can join guest_access rooms",
+   requires => [ local_user_and_room_preparers(), qw( first_api_client ) ],
+
+   do => sub {
+      my ( $user, $room_id ) = @_;
+
+      matrix_set_room_guest_access( $user, $room_id, "can_join" );
+   },
+
+   check => sub {
+      my ( $user, $room_id, $api_client ) = @_;
+
+      register_anonymous_user( $api_client )->then( sub {
```

Comment:
Beware that over the course of running this test, you'll register two anonymous users; one for the pre-check before the `do` block and another one after it.

You might prefer to make a `anonymous_user_preparer()` to share around, so that the test can start off already with an anonymous user. That'll reduce the amount of async indent in the tests, and also mean you'll keep the same user for both `check` blocks.